### PR TITLE
information-schema: add client errors docs

### PR DESCRIPTION
**Pull request by morgo**
### What is changed, added or deleted? (Required)

This adds docs for https://github.com/pingcap/tidb/pull/22382

Currently this is in master and v5.0, but the cherry pick for 4.0 is also pending. It is required for 4.0 to observe a security issue.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: information-schema/information-schema.md [hunk 1]
```diff
@@ -52,6 +52,9 @@ Many `INFORMATION_SCHEMA` tables have a corresponding `SHOW` command. The benefi
 | Table Name                                                                              | Description |
 |-----------------------------------------------------------------------------------------|-------------|
 | [`ANALYZE_STATUS`](/information-schema/information-schema-analyze-status.md)            | Provides information about tasks to collect statistics. |
+| [`CLIENT_ERRORS_SUMMARY_BY_HOST`](/information-schema/client-errors-summary-by-host.md)  | Provides a summary of errors and warnings generated by clients. |
```
Comment (TomShawn) [inline information-schema/information-schema.md:55]: I am a bit confused about how "errors and warnings generated by clients" are also returned to the clients.

Comment (TomShawn) [inline information-schema/information-schema.md:55]:
```suggestion
| [`CLIENT_ERRORS_SUMMARY_BY_HOST`](/information-schema/client-errors-summary-by-host.md)  | Provides a summary of errors and warnings generated by client requests and returned to clients. |
```

File: information-schema/client-errors-summary-by-host.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
+---
+title: CLIENT_ERRORS_SUMMARY_BY_HOST
+summary: Learn about the `CLIENT_ERRORS_SUMMARY_BY_HOST` information_schema table.
+---
+
+# CLIENT_ERROR_SUMMARY_BY_HOST
+
+The table `CLIENT_ERRORS_SUMMARY_BY_HOST` provides a summary of SQL errors and warnings that have been returned to clients, _or_ applications that connect to a TiDB server. These include:
```
[the remark this answers is not in the document]
Comment (morgo) [reply information-schema/client-errors-summary-by-host.md:8]: I think this changes the meaning. I will reword.

File: information-schema/client-errors-summary-by-host.md [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range of duplicate key values.
```
[the remark this answers is not in the document]
Comment (morgo) [reply information-schema/client-errors-summary-by-host.md:12]: It should read: Attempting to insert out-of-range *or*, thanks!

File: information-schema/client-errors-summary-by-host.md [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+summary: Learn about the `CLIENT_ERRORS_SUMMARY_BY_HOST` information_schema table.
+---
+
+# CLIENT_ERROR_SUMMARY_BY_HOST
+
+The table `CLIENT_ERRORS_SUMMARY_BY_HOST` provides a summary of SQL errors and warnings that have been returned to clients that connect to a TiDB server. These include:
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range or duplicate key values.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-host.md:12]: Let's use the noun structure for every item in the list.
```suggestion
* The attempt to insert out-of-range or duplicate key values.
```

File: information-schema/client-errors-summary-by-host.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+* Permission errors.
+* A table does not exist.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-host.md:14]:
```suggestion
* A table that does not exist.
```

File: information-schema/client-errors-summary-by-user.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+summary: Learn about the `CLIENT_ERRORS_SUMMARY_BY_USER` information_schema table.
+---
+
+# CLIENT_ERROR_SUMMARY_BY_HOST
+
+The table `CLIENT_ERRORS_SUMMARY_BY_USER` provides a summary of SQL errors and warnings that have been returned to clients that connect to a TiDB server. These include:
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range or duplicate key values.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-user.md:12]:
```suggestion
* The attempt to insert out-of-range or duplicate key values.
```

File: information-schema/client-errors-summary-by-user.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+
+* `USER`: The authenticated user.
+* `ERROR_NUMBER`: The MySQL-compatible error number that was returned.
+* `ERROR_MESSAGE`: The error message which matches the error number (in prepared statement form).
+* `ERROR_COUNT`: The number of times this error was returned to the user.
+* `WARNING_COUNT`: The number of times this warning was returned to the user.
+* `FIRST_SEEN`: The first time this error (or warning) was sent to the user.
+* `LAST_SEEN`: The most recent time this error (or warning) was sent to the user.
+
+The following example shows a warning being generated when connecting to a local TiDB server. The summary is reset after executing `FLUSH CLIENT_ERRORS_SUMMARY`:
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-user.md:58]:
```suggestion
The following example shows a warning being generated when the client connects to a local TiDB server. The summary is reset after executing `FLUSH CLIENT_ERRORS_SUMMARY`:
```

File: information-schema/client-errors-summary-global.md [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+summary: Learn about the `CLIENT_ERRORS_SUMMARY_GLOBAL` information_schema table.
+---
+
+# CLIENT_ERROR_SUMMARY_BY_HOST
+
+The table `CLIENT_ERRORS_SUMMARY_GLOBAL` provides a global summary of all SQL errors and warnings that have been returned to clients that connect to a TiDB server. These include:
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range of duplicate key values.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-global.md:12]:
```suggestion
* The attempt to insert out-of-range of duplicate key values.
```

File: information-schema/client-errors-summary-global.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+* Permission errors.
+* A table does not exist.
+
+Client errors are returned to the client via the MySQL server protocol, where applications are expected to take appropriate action. The `information_schema`.`CLIENT_ERRORS_SUMMARY_GLOBAL` table provides a high level overview, and is useful in the scenario that applications are not correctly handling (or logging) errors returned by the TiDB server.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-global.md:16]:
```suggestion
Client errors are returned to the client via the MySQL server protocol, where applications are expected to take appropriate action. The `information_schema`.`CLIENT_ERRORS_SUMMARY_GLOBAL` table provides a high-level overview, and is useful in the scenario where applications are not correctly handling (or logging) errors returned by the TiDB server.
```

File: information-schema/client-errors-summary-by-user.md [hunk 10]
```diff
@@ -0,0 +1,87 @@
...
+
+The table `CLIENT_ERRORS_SUMMARY_BY_USER` provides a summary of SQL errors and warnings that have been returned to clients that connect to a TiDB server. These include:
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range or duplicate key values.
+* Permission errors.
+* A table does not exist.
+
+Client errors are returned to the client via the MySQL server protocol, where applications are expected to take appropriate action. The `information_schema`.`CLIENT_ERRORS_SUMMARY_BY_USER` table provides an useful method to inspect errors in the scenario that applications are not correctly handling (or logging) errors returned by the TiDB server.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-user.md:16]:
```suggestion
Client errors are returned to the client via the MySQL server protocol, where applications are expected to take appropriate action. The `information_schema`.`CLIENT_ERRORS_SUMMARY_BY_USER` table provides an useful method to inspect errors in the scenario where applications are not correctly handling (or logging) errors returned by the TiDB server.
```

File: information-schema/client-errors-summary-by-user.md [hunk 11]
```diff
@@ -0,0 +1,87 @@
...
+
+# CLIENT_ERROR_SUMMARY_BY_HOST
+
+The table `CLIENT_ERRORS_SUMMARY_BY_USER` provides a summary of SQL errors and warnings that have been returned to clients that connect to a TiDB server. These include:
+
+* Malformed SQL statements.
+* Division by zero errors.
+* Attempting to insert out-of-range or duplicate key values.
+* Permission errors.
+* A table does not exist.
```
Comment (TomShawn) [inline information-schema/client-errors-summary-by-user.md:14]:
```suggestion
* A table that does not exist.
```

**Review by TomShawn — COMMENTED** [review]
Minor comments

**Pull request merged**